src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs

```diff
@@ -53,66 +35,99 @@ public static partial class Path
-
-        private static bool IsDirectorySeparator(char c)
+        private static bool IsDirectoryOrVolumeSeparator(char c)
         {
-            return c == DirectorySeparatorChar || c == AltDirectorySeparatorChar;
+            return PathInternal.IsDirectorySeparator(c) || VolumeSeparatorChar == c;
         }
 
-        private static bool IsDirectoryOrVolumeSeparator(char c)
+        // Expands the given path to a fully qualified path. 
+        [Pure]
+        [System.Security.SecuritySafeCritical]
+        public static string GetFullPath(string path)
         {
-            return IsDirectorySeparator(c) || VolumeSeparatorChar == c;
+            string fullPath = GetFullPathInternal(path);
+
+            // Emulate FileIOPermissions checks, retained for compatibility
+            PathInternal.CheckInvalidPathChars(fullPath, true);
+
+            int startIndex = fullPath.StartsWith(PathInternal.LongPathPrefix, StringComparison.Ordinal) ? PathInternal.LongPathPrefix.Length + 2 : 2;
+            if (fullPath.Length > startIndex && fullPath.IndexOf(':', startIndex) != -1)
+            {
+                throw new NotSupportedException(SR.Argument_PathFormatNotSupported);
+            }
+
+            return fullPath;
         }
 
-        // Gets the length of the root DirectoryInfo or whatever DirectoryInfo markers
-        // are specified for the first part of the DirectoryInfo name.
-        // 
-        private static int GetRootLength(string path)
+        /// <summary>
+        /// Checks for known bad extended paths (paths that start with \\?\)
+        /// </summary>
+        /// <param name="fullCheck">Check for invalid characters if true.</param>
+        /// <returns>'true' if the path passes validity checks.</returns>
+        private static bool ValidateExtendedPath(string path, bool fullCheck)
         {
-            CheckInvalidPathChars(path);
-
-            int i = 0;
-            int length = path.Length;
+            if (path.Length == PathInternal.LongPathPrefix.Length)
+            {
+                // Effectively empty and therefore invalid
+                return false;
+            }
 
-            if (length >= 1 && IsDirectorySeparator(path[0]))
+            if (path.StartsWith(PathInternal.UNCLongPathPrefix, StringComparison.Ordinal))
             {
-                // handles UNC names and directories off current drive's root.
-                i = 1;
-                if (length >= 2 && (IsDirectorySeparator(path[1])))
+                // UNC specific checks
+                if (path.Length == PathInternal.UNCLongPathPrefix.Length || path[PathInternal.UNCLongPathPrefix.Length] == DirectorySeparatorChar)
                 {
-                    i = 2;
-                    int n = 2;
-                    while (i < length && (!IsDirectorySeparator(path[i]) || --n > 0)) i++;
+                    // Effectively empty and therefore invalid (\\?\UNC\ or \\?\UNC\\)
```

Comment:
Same goes for \\server. You need at least a server\share in either case. In this check you can't have \\server\\share. Windows is very particular about UNCs- they must use backslashes and exactly the right amount (2 & 1) before you get past the server\share. (At that point slashes get normalized.)